gradio/routes.py

```diff
@@ -313,6 +289,9 @@ async def run_predict(
                     content={"error": str(error) if show_error else None},
                     status_code=500,
                 )
+
+            if not (body.batched) and batch:
```

Comment:
This is confusing because shouldn't the predict body always match what's in the config? I think we can get rid of `body.batched`?

Reply:
See above